lore/developers/contributing/designs/network-design.md

```diff
@@ -0,0 +1,76 @@
+
+ - bastion.bonncyi-internal.portbleu.com
+ - logs.bonnyci-internal.portbleu.com
+ - merger01.bonnyci-internal.portbleu.com
+ - nodepool.bonncyi-internal.portbleu.com
+ - zuul.bonnyci-internal.portbleu.com
+
+ * ``nodepool 10.0.0.0/8`` - This is an shared network that will be used by nodepool slaves. None of the static service hosts within the control plane will receive an interface on this network.
+
+Our cloud is also configured with an external network named ``external`` that provides us currently with 30 floating IPs.  These will be assigned to hosts depending on their need, see [Floating IPs](#floating-ips).
+
+### Routers
+
+To allow connectivity between the two isolated networks, a single L3 router will be created.  The router will have a router interface attached for both the ``control-plane`` and ``nodepool`` networks.  Additionally, it will have its router gateway set to provide outbound connectivity through the external network and allow for floating IP association.
+
+### Floating IPs
+
+Some of the hosts running on the control plane will need to be accessed from the outside world for various reasons.  Public addresses for floating IPs will be allocated from the ``external`` network that is available to our cloud, currently providing addresses from the 169.44.171.64/27 range, though our quota currently allows for 50. Each of the following hosts will also be assigned a floating IP from the external network that will be publically accessible on specific ports (see [Security Groups](#security-groups)). Public DNS hostnames will be updated accordingly:
+
+ - bastion.bonnyci-internal.portbleu.com <- contrasjc-bastion.portbleu.com - Provides an ssh entry point from the outside world to the rest of our infrastructure and serves automation logs via http.
+ - logs.bonnyci-internal.portbleu.com <- logs.bonnyci.com - Serves test log output and other test archives via http.
+ - zuul.bonnyci-internal.portbleu.com <- zuul.bonnyci.org - Provides a https endpoint for receiving Github webhook notifications, and the serves Zuul status JSON and web site.
+
+### Security Groups
+
+Security groups will be used to limit the connectivity from the outside world to only specific ports for the services described above.  They will also be used to classify and limit internal connectivity between internal networks.  Traffic on any port not explicitly allowed will be denied.
+
+The following security groups will be created on the ``control-plane`` project:
+
+* ``sg-control-plane`` - This group will allow all traffic from the other hosts that are part of the same ``sg-control-plane`` security group.
+- Ports open: All ports from security group ``sg-control-plane``.
+
+* ``sg-zuul-merger`` - This group will allow git to be served to nodepool slaves.
+- Ports open:  8858 from network 10.0.0.0/8
```

Comment:
Ah. I think the API lets you choose either a specific CIDR or the name of a security group. If we did it by security group, then that doesn't have to change if we update or add more subnets, but then it also requires we put the right security group on the node pool nodes, so I don't know which way is "better".